web/packages/teleport/src/Roles/RoleEditor/StandardEditor/standardmodel.ts

```diff
@@ -1161,8 +1163,13 @@ function optionsToModel(
 
     ...unsupported
   } = options;
-
   for (const key of uneditableOptionKeys) {
+    // delete saml idp option for role v8 and above as it
+    // is no longer supported.
+    if (!isLegacySamlIdpRbac(roleVersion) && key === 'idp') {
+      delete unsupported[key];
+      continue;
```

Comment:
Starting v8, we wont set the idp option at all. So the field is deleted when traversing to and from standard modal. 

For v7 and below, the existing behaviour of `role.options.idp.saml.enabled - true` remains unchanged and any changes to that field will prompt "this role is too complex..." message, as we do it right now.